python/lsst/pipe/tasks/mergeDetections.py

```diff
@@ -169,24 +212,57 @@ def __init__(self, butler=None, schema=None, **kwargs):
 
         @param[in] schema     the schema of the detection catalogs used as input to this one
         @param[in] butler     a butler used to read the input schema from disk, if schema is None
+        @param[in] initInputs This a PipelineTask only argument that holds all inputs passed in
+                              through the PipelineTask middleware
         @param[in] **kwargs   keyword arguments to be passed to CmdLineTask.__init__
 
         The task will set its own self.schema attribute to the schema of the output merged catalog.
         """
-        CmdLineTask.__init__(self, **kwargs)
+        super().__init__(**kwargs)
+        if initInputs is not None:
+            schema = initInputs['schema'].schema
+
         self.makeSubtask("skyObjects")
         self.schema = self.getInputSchema(butler=butler, schema=schema)
 
         filterNames = [getShortFilterName(name) for name in self.config.priorityList]
         filterNames += [self.config.skyFilterName]
         self.merged = afwDetect.FootprintMergeList(self.schema, filterNames)
 
+    def getInitOutputDatasets(self):
+        return {"outputSchema": afwTable.SourceCatalog(self.schema)}
+
     def runDataRef(self, patchRefList):
         catalogs = dict(readCatalog(self, patchRef) for patchRef in patchRefList)
-        mergeCatalog = self.run(catalogs, patchRefList[0])
-        self.write(patchRefList[0], mergeCatalog)
-
-    def run(self, catalogs, patchRef):
+        skyInfo = getSkyInfo(coaddName=self.config.coaddName, patchRef=patchRefList[0])
+        idFactory = self.makeIdFactory(patchRefList[0])
+        skySeed = patchRefList[0].get(self.config.coaddName + "MergedCoaddId")
+        mergeCatalogStruct = self.run(catalogs, skyInfo, idFactory, skySeed)
+        self.write(patchRefList[0], mergeCatalogStruct.outputCatalog)
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        # FINDME: DM-15843 needs to come back and address this function with final solution
+        inputData["skySeed"] = 0
+        inputData["idFactory"] = afwTable.IdFactory.makeSimple()
+        catalogDict = {dataId['abstract_filter']: cat for dataId, cat in zip(inputDataIds['catalogs'],
+                       inputData['catalogs'])}
+        inputData['catalogs'] = catalogDict
+        skyMap = inputData.pop('skyMap')
+        # Can use the first dataId to find the tract and patch being worked on
+        tractNumber = inputDataIds['catalogs'][0]['tract']
+        tractInfo = skyMap[tractNumber]
+        patchInfo = tractInfo.getPatchInfo(inputDataIds['catalogs'][0]['patch'])
+        skyInfo = Struct(
+            skyMap=skyMap,
+            tractInfo=tractInfo,
+            patchInfo=patchInfo,
+            wcs=tractInfo.getWcs(),
+            bbox=patchInfo.getOuterBBox()
+        )
```

Comment:
FYI I've just created a ticket that might eventually help with this: https://jira.lsstcorp.org/browse/DM-16919, in that I'd like to just let you declare something like `PatchInfo` as an input Dataset, and you'd get the `PatchInfo` you want directly instead of having to read the SkyMap and unpack the data ID.  That won't help immediately if you need more than a `PatchInfo` here, but I think a `PatchInfo` should probably hold a reference to its own `TractInfo` and `SkyMap` if it doesn't already, and if we did that we wouldn't need the `SkyInfo` struct to be more than a `PatchInfo`.